File: src/main/java/team4/footwithme/vote/domain/VoteItem.java
package team4.footwithme.vote.domain;

import jakarta.persistence.*;

@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn
@Entity
public class VoteItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long voteItemId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vote_id", nullable = false)
    private Vote vote;

    public VoteItem(Vote vote) {
        this.vote = vote;
        vote.addChoice(this);
    }

    protected VoteItem() {
    }

    public Long getVoteItemId() {
        return this.voteItemId;
    }

    public Vote getVote() {
        return this.vote;
    }
}

File: src/main/java/team4/footwithme/vote/domain/VoteItem.kt
package team4.footwithme.vote.domain

import jakarta.persistence.*

@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn
@Entity
open class VoteItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val voteItemId: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vote_id", nullable = false)
    var vote: Vote? = null
        

    constructor(vote: Vote?) {
        this.vote = vote
        vote!!.addChoice(this)
    }

    protected constructor()
}
